contracts/Donations.sol

```diff
@@ -175,7 +178,7 @@ contract Donations is ERC721, AccessControl {
 
         _burn(_id);
 
-        emit DonationBurned(_id);
+        emit DonationBurned(_id, donationId);
```

Comment:
@0xrin1 I'm still confused, why do we need the `donationId`? From the code, I get a feeling that the `id` should be enough, you can already backtrack to the chain by using the txHash

Reply:
My thinking is that for the DonationMinted event we need to emit the donationId that was generated by the graph when creating the corresponding Donation record because we want to know exactly which YieldClaimed event (and therefore Donation record) had an NFT minted for it

Say one transaction has 100 YieldClaimed events (with to address = treasury, so 100 yields that were claimed as donation) how do we know which NFTs line up with which Donation record?

I don't think we can know unless we pass the donationId into the function so that we can keep track of what was minted for what.

For the DonationBurned event it just makes the graph handler much simpler because the id is the donationId, so we can just `const donationMint = DonationMint.load(event.params.donationId)!`

Otherwise we would have to create a DonationBurned record and then stitch that data together on the backend

Reply:
hummm, I see, you are right 👍